sphinx/util/typing.py

```diff
@@ -178,14 +178,27 @@ def _is_annotated_form(obj: Any) -> TypeGuard[Annotated[Any, ...]]:
     return typing.get_origin(obj) is Annotated or str(obj).startswith('typing.Annotated')
 
 
+def _is_unpack_form(obj: Any) -> bool:
+    """Check if the object is :class:`typing.Unpack` or equivalent."""
+    origin = typing.get_origin(obj)
+    __module__ = getattr(origin, '__module__', None)
+    if __module__ == 'typing':
+        return getattr(origin, '__qualname__', None) == 'Unpack'
+
+    if __module__ == 'typing_extensions':
```

Comment:
And now tests are changed so that the assumption holds. So, we are effectively ignoring the fact that `typing_extensions.Unpack` can be used in 3.11. Maybe I should ask on their repo if it's possible to have the same origin for 3.11.